locator/tablets.cc

```diff
@@ -455,6 +455,34 @@ tablet_map::tablet_map(size_t tablet_count)
     _tablets.resize(tablet_count);
 }
 
+tablet_map::~tablet_map() {
+    if (!_tablets.empty()) {
+        tablet_logger.debug("Destroying tablet_map with {} tablets", _tablets.size());
+    }
+}
+
+tablet_map tablet_map::clone() const {
+    return tablet_map(_tablets, _log2_tablets, _transitions, _resize_decision, _resize_task_info, _repair_scheduler_config);
+}
+
+future<tablet_map> tablet_map::clone_gently() const {
+    tablet_container tablets;
+    tablets.reserve(_tablets.size());
+    for (const auto& t : _tablets) {
+        tablets.emplace_back(t);
+        co_await coroutine::maybe_yield();
+    }
```

Comment:
With 100,000 tablets, and each tablet requiring an allocation if there are more than three replicas, this could indeed stall.

Note: tablet_replica is very wasteful, it consumes 32 bytes for each replica.

Reply:
Cc @tgrabiec

Reply:
We could reduce it to 4 bytes by converting host_id to host_idx (just add an x) that indexes into a table of hosts in topology.